frontend/src/components/event/table/FulfilPurchaseRequestTable.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from "react";
+import { Activity } from "../../../types/activity";
+import {
+    TableContainer,
+    Table,
+    TableHead,
+    TableBody,
+    TableRow,
+    TableProps,
+    useTheme,
+    Paper,
+    TableCell,
+    Typography,
+    Checkbox,
+} from "@mui/material";
+
+interface FulfilPurchaseRequestTableProps extends TableProps {
+    activities: Activity[];
+}
+
+const FulfilPurchaseRequestTable: React.FC<FulfilPurchaseRequestTableProps> = ({
+    activities,
+    ...props
+}) => {
+    const theme = useTheme();
+
+    const [selectedActivities, setSelectedActivities] = useState<string[]>([]);
```

Comment:
Try using `useForm`, it might be challenging, but u will need it moving forwards